src/DataTable/useViewMore.ts

```diff
@@ -5,39 +5,40 @@ import * as React from "react"
  * triggered by a click event.
  */
 const useViewMore = () => {
+  let closeTimeout = 0
   const [viewMorePopup, setViewMorePopup] = React.useState<{ content: string; x: number; y: number } | false>(false)
 
-  React.useEffect(() => {
-    if (!viewMorePopup) {
-      return
-    }
-
-    const close = () => {
-      setViewMorePopup(false)
-    }
-
-    document.addEventListener("click", close)
-    document.addEventListener("contextmenu", close)
-    document.addEventListener("scroll", close)
-
-    return () => {
-      document.removeEventListener("click", close)
-      document.removeEventListener("contextmenu", close)
-      document.removeEventListener("scroll", close)
-    }
-  }, [viewMorePopup])
-
   const openViewMore = React.useCallback(
     (content: string) => (e: React.MouseEvent) => {
-      setViewMorePopup({ content, x: e.clientX, y: e.clientY })
+      window.clearTimeout(closeTimeout)
+      e.stopPropagation()
+      setViewMorePopup({
+        content,
+        x: e.clientX > window.innerWidth / 2 ? e.clientX - 8 : e.clientX + 8,
+        y: e.clientY > window.innerHeight / 2 ? e.clientY - 8 : e.clientY + 8,
+      })
+    },
+    [viewMorePopup, closeTimeout],
+  )
+
+  const close = () => {
+    closeTimeout = window.setTimeout(() => setViewMorePopup(false), 300)
```

Comment:
From testing in the PR preview, 300 feels pretty slow and gives me the feeling that we have some performance issues. Also made try if I still could reach the pop-over to select and copy the value from it. Was left frustrated  since having mouse over the pop-over does not keep the pop-over opened.

If there is no requirement to keep it 300, I'd go with smaller value to make it feel snappier. Would 150ms also work here?

```suggestion
    closeTimeout = window.setTimeout(() => setViewMorePopup(false), 150)
```